src/api/operations.ts

```diff
@@ -181,36 +183,90 @@ export const drawRectangle = (options: {
+    asNumber(xN), asNumber(yN), asNumber(rN || 0), asNumber(wN), asNumber(hN),
+  ];
+  const c = r * (1.0 - KAPPA);
+  return [
     pushGraphicsState(),
     options.graphicsState && setGraphicsState(options.graphicsState),
     options.color && setFillingColor(options.color),
     options.borderColor && setStrokingColor(options.borderColor),
     setLineWidth(options.borderWidth),
     options.borderLineCap && setLineCap(options.borderLineCap),
     setDashPattern(options.borderDashArray ?? [], options.borderDashPhase ?? 0),
-    translate(options.x, options.y),
+    translate(x, y),
     rotateRadians(toRadians(options.rotate)),
     skewRadians(toRadians(options.xSkew), toRadians(options.ySkew)),
-    moveTo(0, 0),
-    lineTo(0, options.height),
-    lineTo(options.width, options.height),
-    lineTo(options.width, 0),
+    moveTo(r, 0),
+    lineTo(w - r, 0),
+    r > 0 ? appendBezierCurve(w - c, 0, w, c, w, r) : null,
+    lineTo(w, h - r),
+    r > 0 ? appendBezierCurve(w, h - c, w - c, h, w - r, h) : null,
+    lineTo(r, h),
+    r > 0 ? appendBezierCurve(c, h, 0, h - c, 0, h - r) : null,
+    r > 0 ? lineTo(0, r) : null,
+    r > 0 ? appendBezierCurve(0, c, c, 0, r, 0) : null,
     closePath(),
-
     // prettier-ignore
     options.color && options.borderWidth ? fillAndStroke()
-  : options.color                      ? fill()
-  : options.borderColor                ? stroke()
-  : closePath(),
+    : options.color                      ? fill()
+    : options.borderColor                ? stroke()
+    : null,
 
     popGraphicsState(),
   ].filter(Boolean) as PDFOperator[];
+};
+
+export const drawLines = (options: {
+  points: { x: number | PDFNumber; y: number | PDFNumber }[],
+  borderWidth: number | PDFNumber;
+  color: Color | undefined;
+  borderColor: Color | undefined;
+  rotate: Rotation;
+  xSkew: Rotation;
+  ySkew: Rotation;
+  closePath: boolean | undefined;
+  lineJoin?: LineJoinStyle;
+  lineCap?: LineCapStyle;
+  dashArray?: (number | PDFNumber)[];
+  dashPhase?: number | PDFNumber;
+  graphicsState?: string | PDFName;
+}) =>
+  [
+    pushGraphicsState(),
+    options.graphicsState && setGraphicsState(options.graphicsState),
+    options.color && setFillingColor(options.color),
+    options.borderColor && setStrokingColor(options.borderColor),
+    setLineWidth(options.borderWidth),
+    options.lineCap && setLineCap(options.lineCap),
+    setDashPattern(options.dashArray ?? [], options.dashPhase ?? 0),
+    setLineJoin(options.lineJoin || LineJoinStyle.Miter),
+    translate(options.points[0].x, options.points[0].y),
+    rotateRadians(toRadians(options.rotate)),
+    skewRadians(toRadians(options.xSkew), toRadians(options.ySkew)),
+    ...(options.points.slice(1).map(p =>
+          lineTo(
+            asNumber(p.x) - asNumber(options.points[0].x),
+            asNumber(p.y) - asNumber(options.points[0].y),
+    ))),
+    // Close only if instructed to
+    ...(options.closePath ? [closePath()] : []),
```

Comment:
It's your project, but I would disagree. Most cases where I would use this function I would neither want nor expect that it would close the path automatically; the only reason I even added that option is so that it would also be a viable replacement for "drawPolygon" in other libraries. The function name is "drawLines" because drawing a number of connected lines is a common operation; if you want it to close path by default then it should be called "drawPolygon", but having it able to not closePath by default seems counter-intuitive.

Your call, but I personally would find the idea that "drawLines" would connect the final point and the first point without being explicitly told to do so very strange.